src/NetTopologySuite.Lab/Hull/ChiShape.cs

```diff
@@ -0,0 +1,151 @@
+        /// </summary>
+        /// <returns>Convex hull if tolerance is higher than the maximum edge length, simple polygon if tolerance is low.</returns>
+        /// <exception>If no concavehull can be generated throws not implemented exception.</exception>
+        public Geometry GetResult()
+        {
+            var subdiv = BuildDelaunay();
+            var tris = ExtractTriangles(subdiv);
+            var hull = ComputeHull(tris);
+            return hull;
+        }
+
+        private static IList<QuadEdgeTriangle> ExtractTriangles(QuadEdgeSubdivision subdiv)
+        {
+            var qeTris = QuadEdgeTriangle.CreateOn(subdiv);
+            return qeTris;
+        }
+
+
+        private Geometry ComputeHull(IList<QuadEdgeTriangle> tris)
+        {
+            var quadEdgeTriangleList = new SortedList<QuadEdge, QuadEdgeTriangle>(new QuadEdgeComparer());
+            foreach (var triangle in tris)
+            {
+                CheckMaxEdgeAndListBorderTriangles(quadEdgeTriangleList, triangle);
+            }
+
+            while (quadEdgeTriangleList.Count != 0)
+            {
+                var triangle = quadEdgeTriangleList.Last();
+                var edge = triangle.Key;
+                Debug.WriteLine(triangle.Value);
+                quadEdgeTriangleList.Remove(edge);
+                if (edge.Length.CompareTo(_tolerance) >= 0 && RemovalGivesRegularPolygon(triangle.Value))
+                {
+                    var neighbours = triangle.Value.GetNeighbours();
+                    triangle.Value.Kill();
+                    foreach (var neighbour in from n in neighbours
+                                              where n != null && n.IsLive()
+                                              select n)
+                    {
+                        Debug.WriteLine(neighbour);
+                        CheckMaxEdgeAndListBorderTriangles(quadEdgeTriangleList, neighbour);
+                    }
+                }
+            }
+            var lineMerger = new LineMerger();
+            var fact = new GeometryFactory();
+            foreach (var triangle in tris)
+            {
+                if (!triangle.IsLive()) continue;
+                for (int i = 0; i < 3; i++)
+                {
+                    if (triangle.GetAdjacentTriangleAcrossEdge(i) == null || !triangle.GetAdjacentTriangleAcrossEdge(i).IsLive())
+                    {
+                        lineMerger.Add(triangle.GetEdge(i).ToLineSegment().ToGeometry(fact));
+                    }
+                }
+            }
+            var mergedLine = (LineString)lineMerger.GetMergedLineStrings().FirstOrDefault();
+            if (mergedLine.IsRing)
+            {
+                var lr = new LinearRing(mergedLine.CoordinateSequence, fact);
+
+                var concaveHull = new Polygon(lr, null, fact);
+
+                return concaveHull;
+            }
+            return mergedLine;
```

Comment:
Documentation says that this is supposed to throw if we can't generate a hull, but this returns a 1-dimensional geometry... which is the correct behavior?

Reply:
Let's just throw an exception and call it a day.